### PR TITLE
[Metrics] Unification of FBeta

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Update to fbeta metric:
* Unification of class and functional backend
* Added ``F1`` metric as it is the most commonly used metric
* Renamed class metric `Fbeta`->`FBeta` and functionals `fbeta_score`->`fbeta`, `f1_score`->`f1` for consistency
* `average` argument now also supports `weighted` and `None` as options
* Made averaging more stable towards NaNs (some edge case when `reduction='micro'`)

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In in short, see following bullet-list:  

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by Borda — COMMENTED** [review]
it is quite hard to see what is move and what is change, but I very welcome this unification, I tried already when it was added :] https://github.com/PyTorchLightning/pytorch-lightning/pull/4043#discussion_r502765473

**Comment by SkafteNicki** [comment]
@Borda yes, I saw your comment and promised long time ago update to the metric.
I have a hard time keeping these unifications PRs small, as it involves moving the old implementation, creating/updating class interface to call functional interface, update docs, update tests ect. so it often involve changes in many files :]

**Review by teddykoker — APPROVED** [review]
Great work as always :)

File: pytorch_lightning/metrics/classification/f_beta.py [hunk 1]
```diff
@@ -51,7 +48,10 @@ class Fbeta(Metric):
 
         average:
             * `'micro'` computes metric globally
-            * `'macro'` computes metric for each class and then takes the mean
+            * `'macro'` computes metric for each class and uniformly averages them
+            * `'weighted'` computes metric for each class and do a weighted-average,
```
Comment (awaelchli) [inline pytorch_lightning/metrics/classification/f_beta.py:52]:
```suggestion
            * `'weighted'` computes metric for each class and does a weighted-average,
```

File: pytorch_lightning/metrics/classification/f_beta.py [hunk 2]
```diff
@@ -110,25 +111,88 @@ def update(self, preds: torch.Tensor, target: torch.Tensor):
...
-        denom = self.beta ** 2 * precision + recall
+    Args:
+        num_classes: Number of classes in the dataset.
+        threshold:
+            Threshold value for binary or multi-label logits. default: 0.5
+
+        average:
+            * `'micro'` computes metric globally
+            * `'macro'` computes metric for each class and uniformly averages them
+            * `'weighted'` computes metric for each class and do a weighted-average,
```
Comment (awaelchli) [inline pytorch_lightning/metrics/classification/f_beta.py:157]:
```suggestion
            * `'weighted'` computes metric for each class and does a weighted-average,
```

File: pytorch_lightning/metrics/classification/f_beta.py [hunk 3]
```diff
@@ -110,25 +111,88 @@ def update(self, preds: torch.Tensor, target: torch.Tensor):
...
+                where each class is weighted by their support (accounts for class imbalance)
+            * `None` computes and returns the metric per class
 
-        return class_reduce(num=num, denom=denom, weights=None, class_reduction='macro')
+        multilabel: If predictions are from multilabel classification.
+        compute_on_step:
+            Forward only calls ``update()`` and return None if this is set to False. default: True
```
Comment (awaelchli) [inline pytorch_lightning/metrics/classification/f_beta.py:163]:
```suggestion
            Forward only calls ``update()`` and returns None if this is set to False. default: True
```

File: pytorch_lightning/metrics/functional/f_beta.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+        actual_positives: torch.Tensor,
+        beta: float = 1.0,
+        average: str = "micro"
+) -> torch.Tensor:
+    if average == "micro":
+        precision = true_positives.sum().float() / (predicted_positives.sum())
+        recall = true_positives.sum().float() / (actual_positives.sum())
+    else:
+        precision = true_positives.float() / (predicted_positives)
+        recall = true_positives.float() / (actual_positives)
```
Comment (awaelchli) [inline pytorch_lightning/metrics/functional/f_beta.py:50]:
```suggestion
        precision = true_positives.sum().float() / predicted_positives.sum()
        recall = true_positives.sum().float() / actual_positives.sum()
    else:
        precision = true_positives.float() / predicted_positives
        recall = true_positives.float() / actual_positives
```

File: pytorch_lightning/metrics/functional/f_beta.py [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+        target: ground-truth labels
+        num_classes: Number of classes in the dataset.
+        beta: Beta coefficient in the F measure.
+        threshold:
+            Threshold value for binary or multi-label logits. default: 0.5
+
+        average:
+            * `'micro'` computes metric globally
+            * `'macro'` computes metric for each class and uniformly averages them
+            * `'weighted'` computes metric for each class and do a weighted-average,
```
Comment (awaelchli) [inline pytorch_lightning/metrics/functional/f_beta.py:89]:
```suggestion
            * `'weighted'` computes metric for each class and does a weighted-average,
```

File: pytorch_lightning/metrics/functional/f_beta.py [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+        pred: estimated probabilities
+        target: ground-truth labels
+        num_classes: Number of classes in the dataset.
+        threshold:
+            Threshold value for binary or multi-label logits. default: 0.5
+
+        average:
+            * `'micro'` computes metric globally
+            * `'macro'` computes metric for each class and uniformly averages them
+            * `'weighted'` computes metric for each class and do a weighted-average,
```
Comment (awaelchli) [inline pytorch_lightning/metrics/functional/f_beta.py:142]:
```suggestion
            * `'weighted'` computes metric for each class and does a weighted-average,
```

**Review by awaelchli — APPROVED** [review]
👍 ⚡

**Review by justusschock — APPROVED** [review]
